### ISSUE TITLE
question

### ISSUE THREAD

**Issue by Trigun87** [tommarek/Mining-fee-remover#1]
this works with the latest claymore version?
and why don't use on port something like 3000:10000 ?

**Comment by tommarek** [comment]
Hey, yup I'm using it with v10.2 without any problems. As for the ranges I'm not a huge fan of filtering such a big range of ports but I think it might be reasonable to collect information from major pools and filter those specific ports. Something like:
`iptables -A OUTPUT -p tcp -m multiport --dports 3333,3344,4000,4444,5005,8008,8888,9999  -j NFQUEUE --queue-num 0`
This contains ethermine, suprnova, nanopool, ethpool, etc.

But I still think the user should be specifying the port based on the pool he uses. Also what's worth noting is that by default Claymore can switch pool when mining the fee so you should run it with `-allcoins 0 -allpools 0` to make sure it mines on the same pool as you.

**Comment by Trigun87** [comment]
on ethos if i write show miner i read this:
DevFee: failed to mine for a long time, something is wrong, "nofee" mode is enabled until successful devfee mining.

i edited a little bit ur code since i use a pool with the login so i need to use the username as eth_wallet 


*edit*
i noticed that if u redirect the fee claymore retry to set the fee at the next loop... 
i think claymore wait some response or something from the fee... maybe u should fake a response?

**Comment by tommarek** [comment]
What pool do you use? Also could you please send me the output of `grep -C 2 Rec /var/log/mining-fee-remover.log`? Do you have `-allcoins 0` and `-allpools 0` in `claymore.stub.conf`?

**Comment by tommarek** [comment]
yeah it sends a login request. I think the problem is that your pool uses login rather than just eth wallet.
on ethermine the eth_submitLogin message looks like this 
`{"worker": "eth1.0", "jsonrpc": "2.0", "params": ["0xclaymores_wallet", "x"], "id": 2, "method": "eth_submitLogin"}`
and I change it to:
`{"worker": "eth1.0", "jsonrpc": "2.0", "params": ["0xmy_wallet", "x"], "id": 2, "method": "eth_submitLogin"}`
the length of the message doesn't change and it works just fine. In your case I suspect the problem is that the logins are not the same length and thus the packet length doesn't match with the real message size. It's possible to change it (it's commented out in the code) but I had some issues with it before so I disabled it as it's not vital for me. Could you please send me the example what your messages look like?

**Comment by Trigun87** [comment]
i use miningrigrentals and redirect it to dwarfpool (with mmr i can change the coin from mmr's site for all my rigs instead of change all local.conf of each rig :-) and since the login is always the same i don't need to edit the antifee too )
i tried different script for remove fee (but all devs don't follow anymore the code :-) )

this is the packet from another script (first login packet of the miner) 
2017-12-19 04:20:41:{"worker": "4fdd9c", "jsonrpc": "2.0", "params": ["Skohan.50974", "x"], "id": 2, "method": "eth_submitLogin"}

i changed ur commented line with
pkt[IP].len = pkt[IP].len + (len(new_payload) -  len(pkt[TCP].payload))
and
 def __init__(self, pool, port, eth_wallet, password, worker_name=None):
        self.pool = pool
        self.port = port
        self.password = password
        self.eth_wallet = eth_wallet

**Comment by tommarek** [comment]
Would be interested in the login message that claymore uses to start mining the fee as well to see what the difference actually is. Anyways if you fix it, please send a pull request and I'd be happy to merge it in :-)

**Comment by Trigun87** [comment]
> 11:19 AM ethos@4fdd9c 192.168.0.18 [378.1 hash] /home/ethos # cat testlogging | grep '"x"' | head
> 2017-12-19 04:20:41:{"worker": "4fdd9c", "jsonrpc": "2.0", "params": ["Skohan.50974", "x"], "id": 2, "method": "eth_submitLogin"}
> 2017-12-19 04:35:49:{"worker": "4fdd9c", "jsonrpc": "2.0", "params": ["Skohan.50974", "x"], "id": 2, "method": "eth_submitLogin"}
> 2017-12-19 04:54:19:{"id":2,"jsonrpc":"2.0","method":"eth_login","params":["0xdE088812A9c5005b0dC8447B37193c9e8b67a1fF","x"]}

same stuff :-D 
and this is after when i was testing the other code

> 2017-12-19 23:16:38:{"id":2,"jsonrpc":"2.0","method":"eth_login","params":["0xc1c427cD8E6B7Ee3b5F30c2e1D3f3c5536EC16f5","x"]}
> [*] BEFORE: {"id":2,"jsonrpc":"2.0","method":"eth_login","params":["0xc1c427cD8E6B7Ee3b5F30c2e1D3f3c5536EC16f5","x"]}
> [*] AFTER: {"id": 2, "jsonrpc": "2.0", "method": "eth_login", "params": ["Skohan.50974", "x"]}
> 2017-12-19 23:16:38:3f3c5536EC16f5","x"]}



this was the code
```
#!/usr/bin/env python

# based on:
# https://stackoverflow.com/questions/27293924/change-tcp-payload-with-nfqueue-scapy?rq=1
# https://github.com/DanMcInerney/cookiejack/blob/master/cookiejack.py

import logging
logging.getLogger("scapy.runtime").setLevel(logging.ERROR)

import nfqueue
from scapy.all import *
import os
import re
from os import path
from datetime import datetime, timedelta
import json
from collections import OrderedDict

#os.environ["PATH"] = "/usr/sbin:/sbin:/usr/bin:/bin"
# https://forum.z.cash/t/about-dev-fees-and-how-to-remove-them/9600/36
os.system('/sbin/iptables -A OUTPUT -p tcp —dport 3000:10000 -j NFQUEUE —queue-num 0')  # for dwarfpool
#os.system('iptables -A OUTPUT -p tcp —dport 9999 -d eth-us-west1.nanopool.org -j NFQUEUE —queue-num 0')
#os.system('iptables -A OUTPUT -p tcp —dport 5000 -j NFQUEUE —queue-num 0')
#os.system('iptables -A INPUT -p tcp —dport 5000 -j NFQUEUE —queue-num 0')

my_eth_address = 'Skohan.50974'

logfile = open('/home/ethos/anti_fee_log.txt', 'w', 0)

def callback(arg1, payload):
  data = payload.get_data()
  pkt = IP(data)

  payload_before = len(pkt[TCP].payload)

  payload_text = str(pkt[TCP].payload)
  # jason
  print("%s:%s" % (datetime.now().strftime('%Y-%m-%d %H:%M:%S'), payload_text))
  if ('ogin' in payload_text):
    json_data=json.loads(payload_text, object_pairs_hook=OrderedDict)
    if json_data['params']:
      if my_eth_address not in json_data['params'][0]:
        print('[*] DevFee Detected - Replacing Address - %s\n' % datetime.now().strftime('%Y-%m-%d %H:%M:%S'))
        print('[*] REPLACED FROM %s TO %s\n' % (json_data['params'][0], my_eth_address))
        json_data['params'][0] = my_eth_address
        print("[*] BEFORE: %s\n" % payload_text)
        print("[*] AFTER: %s\n" % json.dumps(json_data))
        payload_text=json.dumps(json_data) + '\n'
  pkt[TCP].payload = payload_text

  payload_after = len(payload_text)

  payload_dif = payload_after - payload_before

  pkt[IP].len = pkt[IP].len + payload_dif

  pkt[IP].ttl = 40

  del pkt[IP].chksum
  del pkt[TCP].chksum
  payload.set_verdict_modified(nfqueue.NF_ACCEPT, str(pkt), len(pkt))

  logfile.write(payload_text)
  logfile.write('\n')
  logfile.flush()

def main():
  q = nfqueue.queue()
  q.open()
  q.bind(socket.AF_INET)
  q.set_callback(callback)
  q.create_queue(0)
  try:
    q.try_run() # Main loop
  except KeyboardInterrupt:
    q.unbind(socket.AF_INET)
    q.close()
    if path.exists('./restart_iptables'):
      os.system('./restart_iptables')

main()
```

ofc i edited a little bit that too :-D

**Comment by tommarek** [comment]
does that work?

**Comment by Trigun87** [comment]
just try it :-) when i was running that i noticed a spam of fee from claymore

this is the log of the file with  `grep '"x"' `
https://paste.ubuntu.com/26220836/
at the start of the file i was just testing the script without change the pkt
after i enabled the change and claymore was spamming the fee every min instead every hour :-)

**Comment by Nothro** [comment]
I've noticed that the Mining_fee_remover log file stops actually recording after a few minutes.  But it seems to continue doing it's thing.  I'd like to view the progress if possible.  Any suggestions? Thanks in advance.
